[BLAS][portBLAS]Add wait to copy event in ROTMG operator

s-Nick opened the pull request:
# Description

Currently the dependencies on copy event isn't always respected with OpenCL on CPU devices, this cause random failure with this operator. 
To reproduce the failure, run `RotmgUsm` operator in a loop using OpenCL CPU portBLAS backend, it will eventually fail, it generally takes less that 10 iterations.
Adding a `wait` on the `memcpy` momentarily fix the issue while waiting for a proper compile fix.

Possible related issue https://github.com/intel/llvm/issues/14623

# Checklist

## All Submissions

- [x] Do all unit tests pass locally? Attach a log.
[rotmg_amd_log.txt](https://github.com/user-attachments/files/16331757/rotmg_amd_log.txt)
[rotmg_cpu_log.txt](https://github.com/user-attachments/files/16331758/rotmg_cpu_log.txt)
[rotmg_nvidia_log.txt](https://github.com/user-attachments/files/16331759/rotmg_nvidia_log.txt)
[rotmg_pvc_log.txt](https://github.com/user-attachments/files/16331760/rotmg_pvc_log.txt)

- [x] Have you formatted the code using clang-format?


## Bug fixes

- [x] Have you included information on how to reproduce the issue (either in a
      GitHub issue or in this PR)?

s-Nick committed:
Add wait to copy event

Currently the dependencies on copy event isn't always respected with OpenCL
on CPU devices. Adding a wait momentarily fix the issue while waiting
for a proper compile fix.

Rbiessy commented on src/blas/backends/portblas/portblas_level1.cxx:
```diff
@@ -370,10 +370,9 @@ sycl::event rotmg(sycl::queue &queue, real_t *d1, real_t *d2, real_t *x1, real_t
     auto y_d =
         (real_t *)sycl::malloc_device(sizeof(real_t), queue.get_device(), queue.get_context());
-    auto copy_in_event = queue.memcpy(y_d, &y1, sizeof(real_t), dependencies);
+    queue.memcpy(y_d, &y1, sizeof(real_t), dependencies).wait();
```
Can you add a comment here to explain that waiting on this `memcpy` is a work around and link to the intel/llvm issue?

s-Nick replied:
Thank you for the review @Rbiessy.
I added a comment in 7f2c805

s-Nick committed:
Add comment to workaround

Add comment to specify why portBLAS needs this workaround and point to
open issue in intel/llvm

hjabird commented on src/blas/backends/portblas/portblas_level1.cxx:
```diff
@@ -370,10 +370,14 @@ sycl::event rotmg(sycl::queue &queue, real_t *d1, real_t *d2, real_t *x1, real_t
+    // Otherwise event dependencies works fine.
+    // The issue has been reported to intel/llvm project here:
+    // https://github.com/intel/llvm/issues/14623
+    queue.memcpy(y_d, &y1, sizeof(real_t), dependencies).wait();
```
This looks like a bit of a performance killer. Could we only wait on the CPU devices since other devices seem to work?

s-Nick commented:
Closed this PR, because we decided to change approach
